e2e/testcases/helm_sync_test.go

```diff
@@ -136,6 +139,354 @@ func TestPublicHelm(t *testing.T) {
+			resource.MustParse(expectedCPULimit),
+			resource.MustParse(expectedMemoryRequest),
+			resource.MustParse(expectedMemoryLimit)),
+		testpredicates.HasExactlyImage("wordpress", "bitnami/wordpress", "", "sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_USERNAME", "test-user-1"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_EMAIL", "test-user@example.com"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_1", "val1"),
+		testpredicates.DeploymentHasEnvVar("wordpress", "TEST_2", "val2")); err != nil {
+		nt.T.Error(err)
+	}
+	if nt.T.Failed() {
+		nt.T.FailNow()
+	}
+
+	nt.T.Log("Apply the ConfigMap with values to the cluster")
+	cm2 := corev1.ConfigMap{
+		TypeMeta:   configMapTypeMeta,
+		ObjectMeta: configMapObjMeta,
+		Data: map[string]string{"values.yaml": `
+image:
+  digest: sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e
+  pullPolicy: Never
+wordpressUsername: test-user-2
+wordpressEmail: override-this@example.com
+resources:
+  requests:
+    cpu: 150m
+    memory: 250Mi
+  limits:
+    cpu: 1
+    memory: 300Mi
+mariadb:
+  primary:
+    persistence:
+      enabled: false
+service:
+  type: ClusterIP`},
+	}
+	if err := nt.KubeClient.Update(&cm2); err != nil {
+		nt.T.Fatal(err)
+	}
+
+	if err := nt.Watcher.WatchObject(kinds.Deployment(), "my-wordpress", "wordpress",
+		[]testpredicates.Predicate{
+			containerImagePullPolicy("Never"),
+			testpredicates.HasCorrectResourceRequestsLimits("wordpress",
+				resource.MustParse(expectedCPURequest),
+				resource.MustParse(expectedCPULimit),
+				resource.MustParse(expectedMemoryRequest),
+				resource.MustParse(expectedMemoryLimit)),
+			testpredicates.HasExactlyImage("wordpress", "bitnami/wordpress", "", "sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e"),
+			testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_USERNAME", "test-user-2"),
+			testpredicates.DeploymentHasEnvVar("wordpress", "WORDPRESS_EMAIL", "test-user@example.com"),
+			testpredicates.DeploymentHasEnvVar("wordpress", "TEST_1", "val1"),
+			testpredicates.DeploymentHasEnvVar("wordpress", "TEST_2", "val2"),
+		},
+		testwatcher.WatchTimeout(nt.DefaultWaitTimeout*2)); err != nil {
+		nt.T.Error(err)
+	}
+	if nt.T.Failed() {
+		nt.T.FailNow()
+	}
+
+	nt.T.Log("Apply the ConfigMap with values to the cluster with incorrect data key")
+	cm3 := corev1.ConfigMap{
+		TypeMeta:   configMapTypeMeta,
+		ObjectMeta: configMapObjMeta,
+		Data: map[string]string{"something-else.yaml": `
+image:
+  digest: sha256:362cb642db481ebf6f14eb0244fbfb17d531a84ecfe099cd3bba6810db56694e
+  pullPolicy: Never
+`},
+	}
+	if err := nt.KubeClient.Update(&cm3); err != nil {
+		nt.T.Fatal(err)
+	}
+	nt.WaitForRootSyncStalledError(rs.Namespace, rs.Name, "Validation", "KNV1061: RootSyncs must reference ConfigMaps with valid spec.helm.valuesFrom.key")
+	if nt.T.Failed() {
```

Comment:
`WaitForRootSyncStalledError` calls `Fatal` by default, so this check is redundant

Reply:
Fixed